tests/test_dataset.py

```diff
@@ -82,8 +83,8 @@ def get_test_data_and_samples(
     batch_config.setup(distributed_config)
     batch_config.validate()
     samples = {
-        phase: [batch[0] for batch in data.get_iterator(batch_config, phase, consumed_samples=0, num_workers=0)]
-        for phase, samples in samples_per_phase.items()
+        phase: list(data.get_iterator(batch_config, phase, consumed_samples=consumed_samples, num_workers=0))
```

Comment:
This makes the existing tests too complicated. Instead, Please test spans with a small number of separate tests specifically targeting them. (Not sure we need full coverage for all cases, or you could make one complicated test case that indirectly test many classes.)